components/support/nimbus-fml/src/backends/kotlin/gen_structs/enum_.rs

```diff
@@ -31,45 +30,20 @@ impl CodeType for EnumCodeType {
         identifiers::class_name(&self.id)
     }
 
-    /// The language specific expression that gets a value of the `prop` from the `vars` object.
-    fn get_value(&self, oracle: &dyn CodeOracle, vars: &dyn Display, prop: &dyn Display) -> String {
-        format!(
-            "{vars}.getString({prop}, {transform})",
-            vars = vars,
-            transform = self.transform(oracle).unwrap(),
-            prop = identifiers::quoted(prop)
-        )
-    }
-
     /// The name of the type as it's represented in the `Variables` object.
     /// The string return may be used to combine with an indentifier, e.g. a `Variables` method name.
     fn variables_type(&self, _oracle: &dyn CodeOracle) -> VariablesType {
         VariablesType::String
     }
 
     /// A function handle that is capable of turning the variables type to the TypeRef type.
-    fn transform(&self, oracle: &dyn CodeOracle) -> Option<String> {
+    fn create_transform(&self, oracle: &dyn CodeOracle) -> Option<String> {
```

Comment:
* `variables_type` codes for how the `Variables` object in Kotlin thinks about this type. It's used to construct method calls, e.g. `getStringList` and `getBoolMap` or `getInt`.
 * `create_transform` is a method handles or closures for creating a value as it comes out of the `Variables` object. This can be None, since sometimes you don't need to transform it.
 * `merge_transform` is a method handle of closure for merging a value with a default. If this is `None`, then the object is just overwritten.
 * `value_getter` is the method that gets the variable out of the `Variables` object.
 * `value_mapper` is the expression to make a new value. It uses the `create_transform` to make a new value. This might be used recursively for lists or maps.
 * `value_merger` is the expression to merge a value with a default.
 * `property_getter` assembles it all together in an optional chain:

```
variables
  ?.{{ value_getter }}
  ?.{{ value_mapper }}
  ?.{{ value_merger }}
  ?: _default.value
```

Note to self: this should be documented someplace!